### PR TITLE
Remove strict conditions for pasting elements

### LINKED ISSUE #3543: Remove strict conditions from paste handler

### PR DESCRIPTION
**Pull request by ruggi**
Fixes #3543

This PR removes the strict conditions on the paste action, and adds tests for the different paste combinations, and updates the `isAllowedToReparent` helper to allow copying the contents of an inactive conditional branch.

As a side effect of this, the PR also fixes copy-paste for conditional branches which would not satisfy the original conditions (e.g. having a `position: absolute` prop).

### CODE REVIEW

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/editor/actions/actions.spec.browser2.tsx [hunk 1]
```diff
@@ -271,4 +278,675 @@ describe('actions', () => {
...
+      want: string
+    }
+    const tests: Array<PasteTest> = [
+      {
+        name: 'a single element',
+        startingCode: `
+		<div data-uid='aaa'>
+			<div data-uid='bbb'>foo</div>
+			<div data-uid='ccc'>bar</div>
+		</div>
```
Comment (balazsbajorics) [inline editor/src/components/editor/actions/actions.spec.browser2.tsx:296]: minor nitpick, but please replace these tabs with spaces!

Comment (ruggi) [reply editor/src/components/editor/actions/actions.spec.browser2.tsx]: https://github.com/concrete-utopia/utopia/pull/3544/commits/cb196676ee271570559f4f9e2da64eeb6d8f80d7

File: editor/src/components/editor/actions/actions.spec.browser2.tsx [hunk 2]
```diff
@@ -271,4 +278,675 @@ describe('actions', () => {
...
+              importsToAdd: {},
+            },
+          ]
+        },
+        pasteInto: EP.appendNewElementPath(TestScenePath, ['aaa']),
+        want: `
+		<div data-uid='aaa'>
+			<div data-uid='bbb'>foo</div>
+			<div data-uid='ccc'>bar</div>
+			<div data-uid='aab'>foo</div>
```
[the remark this answers is not in the document]
Comment (ruggi) [reply editor/src/components/editor/actions/actions.spec.browser2.tsx]: that's a pasted element, it gets the next available digit (`aa*b*`)

[the remark this answers is not in the document]
Comment (balazsbajorics) [reply editor/src/components/editor/actions/actions.spec.browser2.tsx]: _slaps forehead_ thanks!

File: editor/src/components/editor/actions/actions.spec.browser2.tsx [hunk 3]
```diff
@@ -271,4 +278,675 @@ describe('actions', () => {
...
+      want: string
+    }
+    const tests: Array<PasteTest> = [
+      {
+        name: 'a single element',
+        startingCode: `
+        <div data-uid='aaa'>
+            <div data-uid='bbb'>foo</div>
+            <div data-uid='ccc'>bar</div>
+		</div>
```
Comment (balazsbajorics) [inline editor/src/components/editor/actions/actions.spec.browser2.tsx:296]: image
sorry, you forgot the last lines in this file. Github does not forgive! :D

Comment (ruggi) [reply editor/src/components/editor/actions/actions.spec.browser2.tsx]: https://github.com/concrete-utopia/utopia/pull/3544/commits/eb7f9e96b6915ef078b161c2e5c6ebcb4fa08f5e

**Review by seanparsons — APPROVED** [review]

**Review by balazsbajorics — APPROVED** [review]
nice tests!

**Pull request merged**